### PR TITLE
Obstacles generation

### PR DESCRIPTION
**Pull request by Daniel-ND**

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/leveldata/LevelGenerator.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Random;
+
+/**
+ * Generates obstacles.
+ * In later improvements could base generation on custom parameters.
+ */
+public class LevelGenerator {
+    static private final int MIN_INTERVAL_VALUE = 20; // Min value of interval between obstacles.
+    static private final int MAX_INTERVAL_VALUE = 80; // Max value of interval between obstacles.
+    static private final ObstacleType[] OBSTACLE_OPTIONS = {CACTUS, CACTUS, CACTUS, PALM, PALM, BAT}; // Options of obstacles for generation.
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/leveldata/LevelGenerator.java:31]: line to long. Move the commment above.

Comment (Daniel-ND) [reply app/src/main/java/com/google/droidjump/leveldata/LevelGenerator.java]: Done

**Comment by saimonsaret** [comment]
I left a few comment on your other PR which are also relevant here. Please apply them here first.

File: app/src/main/java/com/google/droidjump/leveldata/InfiniteLevelData.java [hunk 2]
```diff
@@ -17,27 +17,39 @@
...
  */
 public class InfiniteLevelData implements LevelStrategy {
+    private static final ObstacleType[] OBSTACLE_TYPES = ObstacleType.values();
     ObstacleData currentObstacle;
     int baseSpeed;
+    private Map<ObstacleType, Integer> frequencies;
+    private LevelGenerator levelGenerator;
 
     public InfiniteLevelData(int fileId, Resources resources) {
+        frequencies = new HashMap<>();
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/leveldata/InfiniteLevelData.java:39]: nit: it is probably better to create "frequencies" inside "getDataFromFile". This way, it we for some reason want to update the level data, this data will be correctly filled.

File: app/src/main/java/com/google/droidjump/LevelsAdapter.java [hunk 3]
```diff
@@ -50,7 +51,7 @@ public int getCount() {
 
     @Override
     public Object getItem(int index) {
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/LevelsAdapter.java:53]: Not sure if I understand: what is this "getItem" used for?

**Review by saimonsaret — APPROVED** [review]

File: app/src/main/java/com/google/droidjump/LevelsAdapter.java (hunk 3 above)
Comment (Daniel-ND) [reply app/src/main/java/com/google/droidjump/LevelsAdapter.java:53]: Nothing, this method was used in Max's previous PRs, now it's never used.

Comment (saimonsaret) [reply app/src/main/java/com/google/droidjump/LevelsAdapter.java:53]: Can you remove it then completely?

**Review by saimonsaret — APPROVED** [review]

**Pull request merged**